### PR TITLE
Fail build on wildcard imports

### LINKED ISSUE #3: [BUG] Fail build when `*` imports are used
### What is the bug?
Currently `*` imports are being used and should have a way to fail the build if found. 

### How can one reproduce the bug?
_Steps to reproduce the behavior._

### What is the expected behavior?
_A clear and concise description of what you expected to happen._

### What is your host/environment?
_Operating system, version._

### Do you have any screenshots?
_If applicable, add screenshots to help explain your problem._

### Do you have any additional context?
_Add any other context about the problem._

### PR DESCRIPTION
**Pull request by dbwiddis**
Signed-off-by: Daniel Widdis <widdis@gmail.com>

### Description
Fails the build if a regex matching a wildcard import is encountered.

### Issues Resolved
Closes #3  

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by peternied — COMMENTED** [review]
What about pulling in checkstyle to enforce this [check] instead of writing it on your own(https://checkstyle.org/config_imports.html#AvoidStarImport)?

**Comment by owaiskazi19** [comment]
> What about pulling in checkstyle to enforce this [check] instead of writing it on your own(https://checkstyle.org/config_imports.html#AvoidStarImport)?

We have removed CheckStyle from OpenSearch to have one formatting tool(Spotless) throughout the repo and following the same here as well.

**Comment by dbwiddis** [comment]
Ready for review.  [Initial test failure after adding the check](https://github.com/opensearch-project/opensearch-sdk/runs/6942878760?check_suite_focus=true).  Tests pass after removing star imports in c0f119d99d550d1a84ffa02cc7ededc1f428d252

**Comment by dbwiddis** [comment]
Note: the error message isn't very informative and I had to hunt down the star imports; however, for future failures they'll likely be part of the diff and much easier to identify.   [It may be possible to get line numbers](https://stackoverflow.com/questions/27089509/how-to-get-access-to-line-number-in-file-copy-filter-task) but that adds some complexity.

**Comment by peternied** [comment]
> We have removed CheckStyle from OpenSearch to have one formatting tool(Spotless) throughout the repo and following the same here as well.

From https://github.com/opensearch-project/OpenSearch/issues/1362, I see 'it caused issue on plugins side '.  Seems like this is a cause of tight build system coupling rather than the tool itself being a problem.

While I agree that removing * imports is a good thing, lack of built-in IDE integration and a host of features that might need to be duplicated for future coverage will be the cost of this pattern.

File: gradle/formatting.gradle [hunk 1]
```diff
@@ -10,6 +10,13 @@ allprojects {
             eclipse().configFile rootProject.file('formatter/formatterConfig.xml')
             trimTrailingWhitespace()
 
+            custom 'Refuse wildcard imports', {
+                // Wildcard imports can't be resolved; fail the build
+                if (it =~ /\s+import .*\*;/) {
```
Comment (peternied) [inline gradle/formatting.gradle:15]:
```suggestion
                if (it =~ /\s*import.*\*;/) {
```

Comment (dbwiddis) [reply gradle/formatting.gradle:15]: Imports nearly always start after a new line and the `\s+` was in preference to `[\n|\r|\n\r]\s*`.  There should always be a `package` declaration prior to the imports, although I'm not sure we have a style check to enforce declaring a package.

I think `import*\*;` would behave identically to `\s*import.*\*;` and don't think that's what we want.

**Review by peternied — APPROVED** [review]
Keeping this commit unblocked for merging to keep progress moving on this pull request.  If we encounter 'checks' that are already implemented via checkstyle, I think it might be worthwhile to revisit some of these comments.

Thanks for the great back-and-forth @dbwiddis

**Comment by owaiskazi19** [comment]
> From [opensearch-project/OpenSearch#1362](https://github.com/opensearch-project/OpenSearch/issues/1362), I see 'it caused issue on plugins side '. Seems like this is a cause of tight build system coupling rather than the tool itself being a problem.

We wanted to have only one formatting tool for OpenSearch and that's why asked the Plugin teams to follow the same. Does CheckStyle provides more benefits than Spotless? Something we can think of for the future definitely.

**Pull request merged**